openassessment/data.py

```diff
@@ -1577,3 +1579,122 @@ def get_file_uploads(self, missing_blank=False):
                 files.append(file_upload)
             self.file_uploads = files
         return self.file_uploads
+
+
+def score_type_to_string(score_type):
+    """
+    Converts the given score type into its string representation.
+    """
+    SCORE_TYPE_MAP = {
+        PEER_TYPE: "Peer",
+        SELF_TYPE: "Self",
+        STAFF_TYPE: "Staff",
+        }
+    return SCORE_TYPE_MAP.get(score_type, "Unknown")
+
+def parts_summary(assessment_obj):
+    """
+    Retrieves a summary of the parts from a given assessment object.
+    """
+    return [
+        {
+            'type': part.criterion.name,
+            'score': part.points_earned,
+            'score_type': part.option.name if part.option else "None",
+        }
+        for part in assessment_obj.parts.all()
+    ]
+
+def get_scorer_data(anonymous_scorer_id):
+    """
+    Retrieves the grader's data (full name, username, and email) based on their anonymous ID.
+    """
+    scorer_username = map_anonymized_ids_to_usernames([anonymous_scorer_id]).get(anonymous_scorer_id, "Unknown")
+    scorer_name = map_anonymized_ids_to_fullname([anonymous_scorer_id]).get(anonymous_scorer_id, "Unknown")
+    scorer_email = map_anonymized_ids_to_emails([anonymous_scorer_id]).get(anonymous_scorer_id, "Unknown")
```

Comment:
Can you refactor this a bit, so it's easier to view. You can use a something like

```python
map_type = {
  "username": function,
  ...
}

username = map_type("username").get(anonymous_scorer_id, "Unknown")
```

Or something similar

Reply:
In [this PR](https://github.com/openedx/edx-ora2/pull/2076) there's a pretty interesting suggestion about that implementation that I think we should follow

Reply:
Refactor Done